app/screens/manage_channel_members/manage_channel_members.tsx

```diff
@@ -0,0 +1,277 @@
+        if (pIndex !== -1) {
+            const newProfiles = [...profiles];
+            newProfiles.splice(pIndex, 1);
+            setProfiles(newProfiles);
+
+            const newMembers = [...channelMembers];
+            newMembers.splice(mIndex, 1);
+            setChannelMembers(newMembers);
+
+            await NavigationStore.waitUntilScreensIsRemoved(USER_PROFILE);
+            showRemoveChannelUserSnackbar();
+        }
+    }, [profiles, channelMembers]);
+
+    const handleUserChangeRole = useCallback(async ({userId, schemeAdmin}: {userId: string; schemeAdmin: boolean}) => {
+        const clone = channelMembers.map((m) => {
+            if (m.user_id === userId) {
+                m.scheme_admin = schemeAdmin;
+                return m;
+            }
+            return m;
+        });
+
+        setChannelMembers(clone);
+    }, [channelMembers]);
+
+    const data = useMemo(() => {
+        const isSearch = Boolean(term);
+        if (isSearch) {
+            return filterProfilesMatchingTerm(searchResults, term);
+        }
+        return profiles;
+    }, [term, searchResults, profiles]);
+
+    useNavButtonPressed(MANAGE_BUTTON, componentId, toggleManageEnabled, [toggleManageEnabled]);
+
+    useEffect(() => {
+        mounted.current = true;
+        if (canManageAndRemoveMembers) {
+            updateNavigationButtons(false);
+        }
+        getProfiles();
+        return () => {
+            mounted.current = false;
+        };
+    }, [canManageAndRemoveMembers]);
+
+    useEffect(() => {
+        const removeUserListener = DeviceEventEmitter.addListener(Events.REMOVE_USER_FROM_CHANNEL, handleRemoveUser);
+        const changeUserRoleListener = DeviceEventEmitter.addListener(Events.MANAGE_USER_CHANGE_ROLE, handleUserChangeRole);
+        return (() => {
+            removeUserListener?.remove();
+            changeUserRoleListener?.remove();
+        });
+    }, [handleRemoveUser, handleUserChangeRole]);
+
+    return (
+        <SafeAreaView
+            style={styles.container}
+            testID='manage_members.screen'
+        >
+            <View style={styles.searchBar}>
+                <Search
+                    autoCapitalize='none'
+                    cancelButtonTitle={formatMessage({id: 'mobile.post.cancel', defaultMessage: 'Cancel'})}
+                    keyboardAppearance={getKeyboardAppearanceFromTheme(theme)}
+                    onCancel={clearSearch}
+                    onChangeText={onSearch}
+                    onSubmitEditing={search}
+                    placeholder={formatMessage({id: 'search_bar.search', defaultMessage: 'Search'})}
+                    placeholderTextColor={changeOpacity(theme.centerChannelColor, 0.5)}
+                    testID='manage_members.search_bar'
+                    value={term}
+                />
+            </View>
+
+            {/* TODO: https://mattermost.atlassian.net/browse/MM-48830 */}
```

Comment:
It would be nice to have some context on the code. Is easy to go to the ticket, but easier to just read a short context (1 line is more than enough)